server/liveblog/blogs/embeds.py

```diff
@@ -233,7 +231,35 @@ def embed(blog_id, theme=None, output=None, api_host=None):
     if is_amp:
         embed_template = 'embed_amp.html'
 
-    return render_template(embed_template, **scope)
+    response_content = render_template(embed_template, **scope)
+
+    if is_amp and output and theme.get('supportAdsInjection', False):
+        parsed_content = BeautifulSoup(response_content, "lxml")
+        ads = get_advertisements_list(output)
+
+        frequency = output["settings"].get("frequency", 2)
+        order = output["settings"].get("order", 1)
+
+        # let's remove hidden elements initially because they're just garbage
+        # complex validation because `embed` it's also called from outside without request context
+        if not request or request and not request.args.get('amp_latest_update_time', False):
+            hidden_items = parsed_content.find_all('article', class_="hide-item")
```

Comment:
can we have this class from here?
https://github.com/liveblog/liveblog/pull/1100/files#diff-87a22de194f411edf303419a8f678598R52

Reply:
not sure, but I think we can reuse a single variable to avoid issues later. Will do the change